capi/geos_c.h.in

```diff
@@ -2441,6 +2447,26 @@ extern GEOSGeometry GEOS_DLL *GEOSGeom_createCollection(
     GEOSGeometry** geoms,
     unsigned int ngeoms);
 
+/**
+* Release the sub-geometries of a collection for management.
+* by the caller. The input collection remains as an empty collection,
+* that the caller is responsible for destroying. The output geometries
+* are also the responsibility of the caller, as is the containing array,
+* which must be freed with GEOSFree().
+* \param collection The collection that will have its components released.
+* \param ngeoms A pointer to a variable that will be filled with the
+*        size of the output array.
+* \return A newly allocated array of GEOSGeometry pointers.
+* \note The caller is responsible for freeing the returned array
+*       with GEOSFree() and all the elements with GEOSGeom_destroy().
+*       If called with an empty collection, null will be returned
+*       and ngeoms set to zero.
+* \since 3.4
+*/
+extern GEOSGeometry GEOS_DLL ** GEOSGeom_releaseCollection(
+    GEOSGeometry * collection,
```

Comment:
What do you think about

`int GEOSGeom_releaseCollection(GEOSGeometry* buf)`

where the caller comes up with `buf` however they like, and is responsible for ensuring that `buf` through `buf + GEOSGeom_getNumGeometries(g)` is writable?

Reply:
I mean, it makes the API simpler but it doesn't reduce the amount of hand waving, it just moves it somewhere else... I guess I can see it? This version matches up to the creation signature kind of neatly.

Reply:
@dbaston did you mean:
```
int GEOSGeom_releaseCollection(GEOSGeometry * collection, GEOSGeometry** buf);
```

If so, the original proposal seems simpler and less error-prone.

Reply:
> I mean, it makes the API simpler but it doesn't reduce the amount of hand waving, it just moves it somewhere else.

Less about simplicity and more about flexibility, maybe you are working from C++ and you want to do something like

```
std::vector<GEOSGeometry*> geoms;
geoms.resize(GEOSGeom_getNumGeometries(coll));
GEOSGeom_releaseCollection(coll, geoms.data());
```

Or maybe you want to put the elements of multiple geometries in a single vector. Maybe nobody wants to do things like that, I don't know. Just tossing out an idea.